src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1108,16 +1304,51 @@ public void Dispose()
             string name,
             IEnumerable<TimeSeriesOperation.AppendOperation> toAppend,
             string changeVectorFromReplication = null)
+        {
+            _sortedAppends ??= new SortedDictionary<TimeSeriesOperation.AppendOperation, TimeSeriesOperation.AppendOperation>(TimeSeriesOperation.AppendOperationComparer.Instance);
+            try
+            {
+                foreach (var operation in toAppend)
+                {
+                    _sortedAppends[operation] = operation;
+                }
+
+                var holder = new SingleResult();
+
+                return AppendTimestamp(context, documentId, collection, name, _sortedAppends.Keys.Select(ToResult), changeVectorFromReplication);
+
+                SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
+                {
+                    holder.Values = element.Values;
+                    holder.Tag = context.GetLazyString(element.Tag);
+                    holder.Timestamp = element.Timestamp.EnsureUtc().EnsureMilliseconds();
+                    holder.Status = TimeSeriesValuesSegment.Live;
+                    return holder;
+                }
+            }
+            finally
+            {
+                _sortedAppends.Clear();
+            }
+        }
+
+        public string IncrementTimestamp(
+            DocumentsOperationContext context,
+            string documentId,
+            string collection,
+            string name,
+            IEnumerable<TimeSeriesOperation.IncrementOperation> toIncrement)
         {
             var holder = new SingleResult();
 
-            return AppendTimestamp(context, documentId, collection, name, toAppend.Select(ToResult), changeVectorFromReplication);
+            return AppendTimestamp(context, documentId, collection, name, toIncrement.Select(ToResult), 
+                changeVectorFromReplication: null, verifyName: true, incrementResults: true, addNewNameToMetadata: true);
 
-            SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
+            SingleResult ToResult(TimeSeriesOperation.IncrementOperation element)
             {
                 holder.Values = element.Values;
-                holder.Tag = context.GetLazyString(element.Tag);
-                holder.Timestamp = element.Timestamp;
+                holder.Tag = context.GetLazyString(_documentDatabase.DbBase64Id);
```

Comment:
I think it is worth to keep the `LSV` of `_documentDatabase.DbBase64Id` globally..